### ISSUE TITLE
Level 1 of PA2

### ISSUE THREAD

**Issue by lulucopter** [nus-cs2030/2122-s1#288]
I face this issue in level 1 of PA2, and I'm not sure why this is so. Doing next() twice on a Rand.of(seed) works, but assigning Rand.of(seed) to a variable and then calling next() twice results in a different output. Anyone knows why?

![image](https://user-images.githubusercontent.com/76448538/142277284-6d43a804-6917-4852-a4ef-b045db32ff58.png)

**Comment by RussellDash332** [comment]
Probably your "My newSeed is ..." is a potential confounder. Try removing it and see if it's right.

**Comment by ikaajj** [comment]
Hi, I had the same issue! My problem was that I initialised the random() with the seed in the constructor, so in my next() it is always using the same random() created in the constructor, that's why it gives a different output cause in r.next().get() called earlier, u already used the random once. You can try fixing it by initialising a new random with the seed in next() and calling nextInt using the new random created instead!

**Comment by jehousoh** [comment]
Hii, I too had the same issue when implementing my next()! Essentially what i'm assuming you did was to create a `private final Random` variable in your Rand class and assign it to `new Random(seed)` in your constructor. This will cause the issue above as you'll then be calling the same Random variable multiple times in your next() function (eg r`.next().get()` calls it once, then `r.next().next().get()` will use the same Random variable again) which will give a different result. Hope this explains what the cause of the issue is and you should be able to fix it by implementing @ikaajj 's suggestion! Hope this helps~

**Comment by lulucopter** [comment]
Thank you for your suggestions! It had fixed my issue. I was stunned since logically the random object is created in each instance I called `next()`. Tbh I am still unsure why `.next()` calls onto the same random object

**Comment by jehousoh** [comment]
Hmm not sure what you mean did you create a Random variable in your Rand class and call it in your `.next()` method? Or did you initialize a new Random class each time your `.next()` method was called? Since the latter is correct and shouldn't have caused the issue.

**Comment by jehousoh** [comment]
The was I understood it was that the former will cause the issue as each time you call .nextInt() from your Random object, it changes the Random object and as it is stored in your Rand class, the next call of the .next() will result in a different Random object being called.

![random](https://user-images.githubusercontent.com/87851444/142379394-5eb2393a-d504-46ba-9713-27371034f284.png)

**Comment by YH-15** [comment]
It may be because that you are using the same Random() instance all the time. Maybe you can try to create a new Random() in the next() method instead of creating Random() in the Rand class constructor.

**Comment by CalebLee** [comment]
Yup I made this mistake with PA2 as well because I was storing the instance of Random() in Rand. You shouldn't use the same Random() instance to generate the random integers for Rand. Instead, the next() method should pass the next integer into a new instance of Random() and return that. Hope this helps!

**Comment by callmeqimin** [comment]
your get() should return a property which is a int seed that intially is 2030 then the next() will be new random(this.seed).nextInt(), which the return will beome your next seed, and the programme will run and run, but keep all the record of your seed generated
